### PR TITLE
[RHDX-448] [RHDX-424] & increase memory_limit to 512

### PR DESCRIPTION
**Pull request by jordanpagewhite**
RHDX-448: We are no longer using the functionality of these modules, so
we should remove them. Removing these modules should remove a ton of
routes from the router table. After these routes are removed, we should
continue to monitor the recurring issues we've experienced that note DB
queries to the router table.
See: https://projects.engineering.redhat.com/browse/RHDX-444

RHDX-424: This pull request removes a custom module that implements
hamburgscleanest/guzzle-advanced-throttle, a throttle middleware on
Guzzle for Wordpress API requests. This throttle has never been
hit/utilized, at least not to my knowledge, in my ~1.5 years on the
project. However, in the last week, this throttle has been triggered
many times, resulting in some instances where empty content can be cached
on, for example, the homepage. We believe that this is a result of
significantly increased editorial (uncacheable) activity in our Drupal
application, as we prepare for various events.

We also introduce some cache bins, in this pull request, for requests
made to the Wordpress API. Upon visiting a freshly built environment,
Drupal will make whatever Wordpress API and/or Download Manager call(s)
are required to load a given page, and then it will cache the responses.

Lastly, I have increase the PHP memory_limit to 512M because my local
build was running out of memory during the build while trying to import
config changes. You can my last comment on RHDX-448 for a reference.
See: https://projects.engineering.redhat.com/browse/RHDX-448

### JIRA Issue Link
* https://projects.engineering.redhat.com/browse/RHDX-424
* https://projects.engineering.redhat.com/browse/RHDX-448

### Verification Process

* The pull request builds properly and all e2e tests are green

#### As an editor, UX designer or any non-dev role
I can navigate the PR environment and perform all of my typical duties. Critical pages like:
* homepage
* RHEL product pages
* OpenJDK product pages
* Topic pages
* An article
* An event page
* A cheat sheet

all look and function as I would expect.

#### As a developer

* Perform the non-dev tasks above
* I can fetch this branch to my local environment, build it successfully and I can, for example, load the homepage and see entries in the `lightning_cache_wordpress_api` table that represent the Wordpress content that was fetched to render that page.
* I can verify cache hits by debugging my code (you’ll have to use something like `var_dump()` and/or `exit()` since we don’t have XDebug right now). See `./web/modules/custom/rhd_assemblies/src/Services/WordpressApi.php`

### CODE REVIEW

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
rebuild this please

File: _docker/drupal/dev/docker-compose.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 version: '2'
 services:
   update-composer:
-    image: images.paas.redhat.com/rhdp/developer-base:rhel-76.3
+    image: images.paas.redhat.com/rhdp/developer-base:rhel-76.4
```
Comment (jordanpagewhite) [inline _docker/drupal/dev/docker-compose.yml:4]: I need to understand if this change is alright based upon Rob Blake's recent PR to update the base image version elsewhere in the codebase

File: _docker/drupal/dev/run-drupal.sh [hunk 2]
```diff
@@ -25,7 +25,7 @@ if [[ $REGISTRY_REDHAT_IO_SKIP_LOGIN != true ]]; then
   docker login registry.redhat.io -u "${REGISTRY_REDHAT_IO_USERNAME}" -p "${REGISTRY_REDHAT_IO_PASSWORD}"
 fi
 docker pull docker-registry.upshift.redhat.com/developers/drupal-data:latest
-docker pull images.paas.redhat.com/rhdp/developer-base:rhel-76.3
+docker pull images.paas.redhat.com/rhdp/developer-base:rhel-76.4
```
Comment (jordanpagewhite) [inline _docker/drupal/dev/run-drupal.sh:28]: I need to understand if this change is alright based upon Rob Blake's recent PR to update the base image version elsewhere in the codebase

File: _docker/drupal/managed-platform/etc/opt/rh/rh-php72/php.d/50-developer-drupal.ini [hunk 3]
```diff
@@ -5,9 +5,9 @@
 ;
 ; Increase the default memory limit for Drupal: https://www.drupal.org/docs/7/managing-site-performance-and-scalability/changing-php-memory-limits
 ;
-memory_limit = 256M
+memory_limit = 512M
```
Comment (jordanpagewhite) [inline _docker/drupal/managed-platform/etc/opt/rh/rh-php72/php.d/50-developer-drupal.ini:8]: I need to ask Rob Coffman what the implications of changing this value could be

**Comment by jordanpagewhite** [comment]
You can also, quickly, review the memcache interactions by enabling the "Show memcache statistics at the bottom of each page" on https://developer-preview-3560.ext.us-west.dc.preprod.paas.redhat.com/admin/config/system/memcache. I just enabled it for testing a moment ago.

File: _docker/drupal/dev/docker-compose.yml (hunk 1 above)
Comment (jordanpagewhite) [reply _docker/drupal/dev/docker-compose.yml:4]: Rob confirmed that these should be updated to 76.4

File: _docker/drupal/dev/run-drupal.sh (hunk 2 above)
Comment (jordanpagewhite) [reply _docker/drupal/dev/run-drupal.sh:28]: Rob confirmed that these should be updated to 76.4

File: _docker/drupal/managed-platform/etc/opt/rh/rh-php72/php.d/50-developer-drupal.ini (hunk 3 above)
Comment (jordanpagewhite) [reply _docker/drupal/managed-platform/etc/opt/rh/rh-php72/php.d/50-developer-drupal.ini:8]: Rob and Jason confirmed that this is not an issue in DAT chat

**Comment by jordanpagewhite** [comment]
I am going to run a series of PHP scripts (using drush) against this PR 3560's Drupal container and against 3538's Drupal container (control) in an attempt to get some measurable difference in the performance of the methods I've updated in this PR. Going forward on this project, we should likely look into a tool (perhaps like JMeter which Jason Smith brought up) to provide a much, much better analysis, but I will do this in the meantime or until I determine a better method of testing these changes.

#### getContentById()

`./vendor/bin/drush php-eval '$start = microtime(true); \Drupal::service("rhd_assemblies.wordpress_api")->getContentById("613047"); print microtime(true) - $start;'`

##### 3560

* 0.10599708557129
* 0.13697290420532
* 0.10386204719543
* 0.11787700653076
* 0.020309925079346

##### 3538

* 0.55933713912964
* 0.33303308486938
* 0.38192796707153
* 0.22562694549561
* 0.6364529132843

#### getContentByCategory()

`./vendor/bin/drush php-eval '$start = microtime(true); \Drupal::service("rhd_assemblies.wordpress_api")->getContentByCategory(["135"], 6, [0 => ["value" => "0"]]); print (microtime(true) - $start) . "\n";'`

##### 3560

* 2.5276689529419 (uncached)
* 0.014634847640991
* 0.088095903396606
* 0.086847066879272
* 0.052729845046997

##### 3538

* 2.51389503479 (uncached)
* 0.23125410079956
* 0.70005202293396
* 0.21450400352478
* 0.23099613189697

#### getCategories()

##### 3560

* 0.019292831420898
* 0.1033878326416
* 0.29555416107178
* 0.12878799438477
* 0.091035842895508

##### 3538

* 0.22787594795227
* 0.28373289108276
* 0.20331406593323
* 0.28693199157715
* 0.20106911659241

**Comment by jordanpagewhite** [comment]
#### PR3538 memcached misses

Here is example of some memcache misses that I am seeing on every page load on PR3538:

```
1.87 | getMulti | cache | drupal_preview-3538_%3Adata%3A-route_provider.route_load%3Aa8667b4c5b8f254f4c1087639a13297272a85ade8a1ac04c894c9527aba6a8c6e430d89bdc5767014f3595ce5005d40bcbaa6e32975f777b9f18b26867fdf9e5 | miss
-- | -- | -- | -- | --
523.37 | set | cache | drupal_preview-3538_%3Adata%3A-route_provider.route_load%3Aa8667b4c5b8f254f4c1087639a13297272a85ade8a1ac04c894c9527aba6a8c6e430d89bdc5767014f3595ce5005d40bcbaa6e32975f777b9f18b26867fdf9e5 | miss
```

whereas in this PR 3560, I'm not seeing that set on every page load, presumably because it is actually set properly and can be fetched in that initial get.

This might validate Jason Smith's assumption that Memcache cannot write the object because it is >1mb.

File: _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/rhd_assemblies.updates.inc [hunk 4]
```diff
@@ -1,14 +1,12 @@
 <?php
 
-use Drupal\Core\Database\Connection;
-use Drupal\Core\Database\Database;
-use Drupal\assembly\Entity\Assembly;
```
Comment (jordanpagewhite) [inline _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/rhd_assemblies.updates.inc:5]: phpcs unused use statements

File: _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/rhd_assemblies.updates.inc [hunk 5]
```diff
@@ -1,14 +1,12 @@
...
-
 /**
  * @file
- * hook_update_N()s for assemblies or related functionality.
+ * Hook_update_N()s for assemblies or related functionality.
```
Comment (jordanpagewhite) [inline _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/rhd_assemblies.updates.inc:5]: phpcs

File: _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/rhd_assemblies.updates.inc [hunk 6]
```diff
@@ -26,7 +24,8 @@ function rhd_assemblies_update_8001() {
     throw new DrupalUpdateException('rhd_assemblies_update_8001 failed to apply. Could not update "assembly-dark" visual style to "dark" in assembly__visual_styles.');
   }
 
-  // Update assembly-dark visual styles to dark in assembly_revision__visual_style.
```
Comment (jordanpagewhite) [inline _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/rhd_assemblies.updates.inc:29]: phpcs

File: _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/rhd_assemblies.updates.inc [hunk 7]
```diff
@@ -46,7 +45,8 @@ function rhd_assemblies_update_8001() {
     throw new DrupalUpdateException('rhd_assemblies_update_8001 failed to apply. Could not update "product-hero--dark" visual style to "dark" in assembly__visual_styles.');
   }
 
-  // Update product-hero--dark visual styles to dark in assembly_revision__visual_style.
```
Comment (jordanpagewhite) [inline _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/rhd_assemblies.updates.inc:49]: phpcs

File: _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/rhd_assemblies.updates.inc [hunk 8]
```diff
@@ -56,7 +56,8 @@ function rhd_assemblies_update_8001() {
     throw new DrupalUpdateException('rhd_assemblies_update_8001 failed to apply. Could not update "product-hero--dark" visual style to "dark" in assembly_revision__visual_styles.');
   }
 
-  // Update product-hero--light visual styles to light in assembly__visual_style.
```
Comment (jordanpagewhite) [inline _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/rhd_assemblies.updates.inc:59]: phpcs

File: _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/rhd_assemblies.updates.inc [hunk 9]
```diff
@@ -66,7 +67,8 @@ function rhd_assemblies_update_8001() {
     throw new DrupalUpdateException('rhd_assemblies_update_8001 failed to apply. Could not update "product-hero--light" visual style to "light" in assembly__visual_styles.');
   }
 
-  // Update product-hero--light visual styles to light in assembly_revision__visual_style.
```
Comment (jordanpagewhite) [inline _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/rhd_assemblies.updates.inc:69]: phpcs

File: _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/rhd_assemblies.updates.inc [hunk 10]
```diff
@@ -86,7 +88,8 @@ function rhd_assemblies_update_8001() {
     throw new DrupalUpdateException('rhd_assemblies_update_8001 failed to apply. Could not update "product-hero--slim" visual style to "assembly-slim" in assembly__visual_styles.');
   }
 
-  // Update product-hero--slim visual styles to slim in assembly_revision__visual_style.
```
Comment (jordanpagewhite) [inline _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/rhd_assemblies.updates.inc:89]: phpcs

File: _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/rhd_assemblies.updates.inc [hunk 11]
```diff
@@ -126,7 +129,8 @@ function rhd_assemblies_update_8001() {
     throw new DrupalUpdateException('rhd_assemblies_update_8001 failed to apply. Could not update "assembly-center" visual style to "centered" in assembly__visual_styles.');
   }
 
-  // Update assembly-center visual styles to centered in assembly_revision__visual_style.
```
Comment (jordanpagewhite) [inline _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/rhd_assemblies.updates.inc:129]: phpcs

File: _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/rhd_assemblies.updates.inc [hunk 12]
```diff
@@ -136,7 +140,8 @@ function rhd_assemblies_update_8001() {
     throw new DrupalUpdateException('rhd_assemblies_update_8001 failed to apply. Could not update "assembly-center" visual style to "centered" in assembly_revision__visual_styles.');
   }
 
-  // Update product-signup-hero--dark visual styles to dark in assembly__visual_style.
```
Comment (jordanpagewhite) [inline _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/rhd_assemblies.updates.inc:139]: phpcs

File: _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/rhd_assemblies.updates.inc [hunk 13]
```diff
@@ -146,7 +151,8 @@ function rhd_assemblies_update_8001() {
     throw new DrupalUpdateException('rhd_assemblies_update_8001 failed to apply. Could not update "product-signup-hero--dark" visual style to "dark" in assembly__visual_styles.');
   }
 
-  // Update product-signup-hero--dark visual styles to dark in assembly_revision__visual_style.
```
Comment (jordanpagewhite) [inline _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/rhd_assemblies.updates.inc:149]: phpcs

File: _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/rhd_assemblies.updates.inc [hunk 14]
```diff
@@ -166,7 +172,8 @@ function rhd_assemblies_update_8001() {
     throw new DrupalUpdateException('rhd_assemblies_update_8001 failed to apply. Could not update "events-hero--light" visual style to "light" in assembly__visual_styles.');
   }
 
-  // Update events-hero--light visual styles to light in assembly_revision__visual_style.
```
Comment (jordanpagewhite) [inline _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/rhd_assemblies.updates.inc:169]: phpcs

File: _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/rhd_assemblies.updates.inc [hunk 15]
```diff
@@ -220,6 +227,27 @@ function rhd_assemblies_update_8001() {
     $assembly->save();
   }
 
-  //  If all updates were performed without exception, return a success message.
```
Comment (jordanpagewhite) [inline _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/rhd_assemblies.updates.inc:223]: phpcs

File: _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/src/Service/DownloadManagerApi.php [hunk 16]
```diff
@@ -114,8 +114,8 @@ public function getContentById($id) {
           $request = $this->client->request('GET', $feed_url);
           $response = $request->getBody()->getContents();
           $data = json_decode($response);
-          // Persist this data to a download_manager_api cache bin for an hour.
-          $this->cacheBin->set($cid, $data, time() + 3600);
```
Comment (jordanpagewhite) [inline _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/src/Service/DownloadManagerApi.php:118]: Swap the 'hard coded' 3600 for the `$this->configFactory->get('redhat_developers')->get('downloadManager.ttl')` config variable

File: _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/src/Service/RemoteContentApiInterface.php [hunk 17]
```diff
@@ -3,12 +3,18 @@
 namespace Drupal\rhd_assemblies\Service;
```
Comment (jordanpagewhite) [inline _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/src/Service/RemoteContentApiInterface.php:3]: phpcs fixes and setting a reasonable class comment

File: _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/src/Service/DownloadManagerApi.php (hunk 16 above)
Comment (staceymosier) [reply _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/src/Service/DownloadManagerApi.php:118]: Is it ever possible for $ttl to not have a value?  Probably unlikely, do we set fallback values when looking up configuration?

Comment (jordanpagewhite) [reply _docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/src/Service/DownloadManagerApi.php:118]: Hmm. It wouldn't hurt to set a fallback. There is always a chance that a mistake was made setting the config variable in one of the `drupal/managed-platform/ansible/templates/[ENVIRONMENT]/drupal/rhd.settings.php.j2 ` files and/or that someone could accidentally remove one at some point in the future. I will make this change now. Thanks!

**Comment by jordanpagewhite** [comment]
rebuild this please

**Comment by jordanpagewhite** [comment]
I believe that the checksum will be zero since we are not using cache tags to invalidate this content, since it is fetched from the Wordpress API

**Pull request merged**